tests/test_scenarios.py

```diff
@@ -0,0 +1,173 @@
+        {"expected_len_triples": "19", "mode": "nostop"},
+    ],
+    "dereference_test2_sparql.yml": [
+        {"expected_len_triples": "28", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+    "dereference_test3_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+    "dereference_test4_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+    "dereference_test5_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+}
+
+
+# test if all objects can be retrieved
+@pytest.mark.usefixtures("httpd_server_base", "all_extensions_testset")
+def test_inputs(httpd_server_base: str, all_extensions_testset):
+    assert httpd_server_base
+
+    for input in Path(INPUT).glob("*"):
+        log.debug(f"{input=}")
+        # get name of file
+        name_file = input.name
+        url = f"{httpd_server_base}{name_file}"
+        log.debug(f"{url=}")
+        req = requests.get(url)
+        assert req.ok
+        ctype = req.headers.get("content-type")
+        clen = int(req.headers.get("content-length"))
+        assert clen > 0
+        log.debug(f"{clen=}")
+        log.debug(f"{ctype=}")
+
+        g = Graph().parse(url)
+        # ttl = g.serialize(format="turtle").strip()
+        # log.debug(f"{ttl=}")
+        log.debug(f"{len(g)=}")
+
+
+# launch a server with subprocess from local_server.py
+@pytest.mark.usefixtures("httpd_server_base", "store_info_sets")
+def test_scenarios(
+    httpd_server_base: str,
+    store_info_sets,
+):
+    assert httpd_server_base
+
+    for config in Path(CONFIGS).glob("*.yml"):
+        if config.name not in SCENARIOS_OUTCOMES:
+            log.debug(f"Skipping scenario {config}")
+            continue
+        for test_case in SCENARIOS_OUTCOMES[config.name]:
+            mode = test_case["mode"]
+            log.debug(f"{mode=}")
+
+            # TODO: when mode is added in 0.0.5 add tests here
+            # for now continue
+
+            if mode == "nostop":
+                log.debug(f"Skipping scenario {config} with mode {mode}")
+                continue
+
+            for store_info in store_info_sets:
+                log.debug(f"Running scenario {config} with store {store_info}")
+                travharv = TravHarv(
+                    config,
+                    store_info,
+                )
+                travharv.process()
+                # assert travharv.error_occurred
+                # other assertions here based on the scenarios
+
+                # make connection with the store
+                # create RDF_store
+                core_store: RDFStore = create_rdf_store(*store_info)
+                rdf_store_access = RDFStoreAccess(core_store)
```

Comment:
very strange these two store vars (store an decorator) are created here?

if you actually need that kind of access, then it should just be accessible through member of `travharv` variable?
what added value for this test to re-instantiate?

Reply:
using only one store now, the one from travharv itself, in any case a new store must be able to retrieve the same info I put in the store. That was the reasoning in my head when I wrote a seperate store access for the remote store.